Ghidra/Processors/8048/data/languages/8048.slaspec

```diff
@@ -0,0 +1,349 @@
+# sleigh specification file for Intel 8048
+
+
+# Do not take BS into account when decompiling
+@define SINGLE_REGISTER_BANK
+
+# Treat R0-R7 as not memory mapped (implies SINGLE_REGISTER_BANK)
+@define INTERNAL_REGISTERS
+
+
+@ifdef INTERNAL_REGISTERS
+@define RegType variables
+@define SINGLE_REGISTER_BANK
+@else
+@define RegType names
+@endif
+
+
+define endian=little;
+define alignment=1;
+
+define space CODE type=ram_space size=2 default;
+define space INTMEM type=ram_space size=1;
+define space EXTMEM type=ram_space size=1;
+define space PORT type=ram_space size=1;
+
+define space register type=register_space size=1;
+
+define register offset=0x00 size=1 [ A SP ];
+@ifdef INTERNAL_REGISTERS
+define register offset=0x10 size=1 [ R0 R1 R2 R3 R4 R5 R6 R7 ];
+@endif
+define register offset=0x20 size=2 [ PC ];
+define register offset=0x30 size=1 [ C AC F0 F1 BS DFB ];  # single bit
+
+
+################################################################
+# Tokens
+################################################################
+
+define token opbyte (8)
+   opfull   = (0,7)
+   oplo     = (0,3)
+   ophi     = (4,7)
+   rn       = (0,2) dec
+   rnfill   = (3,3)
+   ri       = (0,0) dec
+   rifill   = (1,3)
+   opaddr   = (5,7)
+   addrfill = (4,4)
+   pp       = (0,1) dec
+   xpp      = (0,1) dec
+   ppfill   = (2,3)
+   abit     = (5,7) dec
+   abfill   = (4,4)
+   dfb      = (4,4)
+   bs       = (4,4)
+;
+
+define token aopword (16)
+   aoplo     = (0,3)
+   aaddrfill = (4,4)
+   aopaddr   = (5,7)
+   adata     = (8,15)
+;
+
+define token ImmedByte (8)  data=(0,7);
+define token AddrOne (8)    addr8=(0,7);
+
+attach $(RegType) rn [ R0 R1 R2 R3 R4 R5 R6 R7 ];
```

Comment:
The Sleigh Editor also doesn't currently like define variables in the attach.  So I reworked this as well.

@ifdef INTERNAL_REGISTERS
attach variables rn [ R0 R1 R2 R3 R4 R5 R6 R7 ];
attach variables ri [ R0 R1 ];
@else
attach names rn [ R0 R1 R2 R3 R4 R5 R6 R7 ];
attach names ri [ R0 R1 ];
@endif